### PR TITLE
refactor: Remove case/contact "joined" ids from data pulls [CHI-2661]

### PR DESCRIPTION
**Pull request by GPaoloni**
REVIEW ONLY AFTER https://github.com/techmatters/hrm/pull/606 IS MERGED.

## Description
This PR removes the following relationships between entities in the data pulls:
- Profile <> Contacts: this can be instead queried using the contact records.
- Profile <> Cases: this can be instead queried using the contact records -> case records.
- ~Case <> Contacts: this can be instead queried using the contact records.~

The purpose of this cleanup is to remove relationships that may fall out of sync. The contact will always be the source of truth, and if any change takes place on a contact record, it will trigger a re-export, keeping the data updated.

### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-2661)
- [ ] New tests added
- [ ] Feature flags / configuration added

### AFTER YOU MERGE

1. Cut a release tag using the GitHub workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P

### CODE REVIEW

File: hrm-domain/scheduled-tasks/hrm-data-pull/pull-cases.ts [hunk 1]
```diff
@@ -51,15 +51,7 @@ export const pullCases = async (startDate: Date, endDate: Date) => {
     };
   });
 
-  const mapContactsToId = (contacts: Required<{ id: number }>[]) =>
```
Comment (nickhurlburt) [inline hrm-domain/scheduled-tasks/hrm-data-pull/pull-cases.ts:54]: @GPaoloni I think we need to keep the case -> contacts references for now, until we've confirmed with KHP that they're not relying on them already.

Comment (GPaoloni) [reply hrm-domain/scheduled-tasks/hrm-data-pull/pull-cases.ts]: Thanks for pointing out, just reverted that change.

**Comment by murilovmachado** [comment]
@GPaoloni #606 is approved. Let me know when that's merged so I can review this one.

**Review by murilovmachado — APPROVED** [review]
Looks good to me!

I've crossed this out from the description:
_"Case <> Contacts: this can be instead queried using the contact records"_

(Just so we're not confused with the final code changes)

**Pull request merged**